### ISSUE TITLE
The error message when `clientViaApplicationDefaultCredentials` fallbacks to reading `metadata` servers (and failing) is cryptic

### ISSUE THREAD

**Issue by rrousselGit** [google/googleapis.dart#572]
Context for this issue: https://github.com/invertase/dart_firebase_admin/issues/6

It appears that if a user tries to create a client with `clientViaApplicationDefaultCredentials`, yet have neither the environment nor `gcloud/application_default_credentials.json` setup, then `google_apis_auth` fallbacks to reaching out to `metadata.google.internal`.

The problem is that if that endpoint is unavailable (which is likely the case if the user made a mistake and _did_ want to login using the other options), then we end-up with a "Failed host lookup: 'metadata.google.internal'" error.

This is likely confusing, and a better error message could make it more intuitive to understand what the issue is.

**Comment by kevmoo** [comment]
Yup. A good place to do a catch and rethrow w/ something more helpful!

**Comment by rrousselGit** [comment]
Would you be open to have a PR for this?

**Comment by kevmoo** [comment]
> Would you be open to have a PR for this?

ABSOLUTELY!
